### ISSUE TITLE
Single scalar ground truth for a batch?

### ISSUE THREAD

**Issue by riaqn** [tensorflow/tensorflow#51090]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.5.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
When calling `model.fit(X)` where `X` is a generator giving the input and ground truth of batches, the ground truth has to be a vector (otherwise it emits error). However, in some case, I want to train a model with a single scalar ground truth for a batch. I would like to aggregate the prediction of the batch into a single scalar and compare it with the single scalar ground truth.  

**Will this change the current api? How?**
It would make the API more permissible. `fit()` would allow the ground truth for a batch to be a single scalar.

**Who will benefit with this feature?**
Those want to train a model with a single scalar ground truth for a batch. 

**Any Other info.**
I don't see why this is not allowed. The ground truth is only used in loss calculation, and it should be fine to be a single scalar as long as the user loss function can handle that.
Currently I simply zero-pad `y_true` into a vector and only use `y_true[0]` in loss calculation.

**Comment by rmothukuru** [comment]
@riaqn,
Can you please explain the Use Case where you think this feature would be useful? Thanks!

**Comment by riaqn** [comment]
@rmothukuru Thank you for taking care of this. I'm training a model that inputs image and outputs some rating of that image. To augment my datasets, instead of labeling each image with a rating, I label a video excerpt with a single rating.  In a single batch, images will be frames from a single video excerpt, and the ground truth will be a single scalar representing the rating of that video excerpt. In `loss` calculation, I would do some sophisticated calculation (not just `reduce_mean`) to aggregate the `y_pred` of each frame into a single scalar,  and compare that with `y_true` which is supposed to be a single scalar.

**Comment by jvishnuvardhan** [comment]
@riaqn Can you please share a simple standalone code to reproduce the issue or add more details about your use case? Are you going to have a batch of data with single label? Are not shuffling the data? I am not sure how it is going to perform.

If you add little more details and a small reproducible code, it would help us understand the feature better.

If there is any actionable PRs, please feel free to open them in [keras-team/keras](https://github.com/keras-team/keras/issues) repository. 

Please note that Keras development moved to keras-team/keras repository to focus entirely on only keras. Thanks!

**Comment by riaqn** [comment]
@jvishnuvardhan Thank you for the reply. Unfortunately I have switched to pytorch now. Please feel free to close the issue. Thank you.
